shell.c

```diff
@@ -1,17 +1,25 @@
 #include "emulator.h"
+#include <SDL2/SDL.h>
 #include <ctype.h>
+#include <math.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 
+#define CLOCK_SPEED_MS 2000
+#define TICK (1000 * (1.0 / 60.0))
+#define CYCLES_PER_TICK (CLOCK_SPEED_MS * TICK)
+
+int run_cpu(i8080 *cpu, int cycles);
+int pflag = 0;
+int dflag = 0;
+int opt;
```

Comment:
Does opt need to be global? I believe it's only ever used in main while calling getopt.

Reply:
I think that's correct. I'll move it back.